shared/rakefiles/scripts/couchdb_backup_restore.sh

```diff
@@ -0,0 +1,260 @@
+  print "Checking membership status"
+  retries=0
+  nodes=0
+  while [ "${retries}" -lt "${MAX_RETRIES}" ] && [ "${nodes}" != "${COUCHDB_REPLICAS}" ]; do
+
+    # shellcheck disable=SC2016
+    nodes="$(kubectl exec -n "${COUCHDB_NAMESPACE}" -it "${COUCHDB_STATEFULSET_NAME}-0" -c couchdb -- \
+      sh -c 'curl -s http://$COUCHDB_USER:$COUCHDB_PASSWORD@127.0.0.1:5984/_membership' | jq -re '.cluster_nodes | length')"
+    [ "${nodes}" != "${COUCHDB_REPLICAS}" ] && printf '.' && sleep "${SLEEP}"
+
+    retries=$((retries + 1))
+  done
+  # shellcheck disable=SC2015
+  [ "${nodes}" = "${COUCHDB_REPLICAS}" ] && printf '\n' || fail "Time-out while waiting for CouchDB membership" 
+}
+
+
+# Init helper functions
+check_colors
+
+# Check prerequisites
+print_header "Checking prerequisites"
+verify_binaries
+
+# Read num of Couch replicas
+print_header "Reading current set of replicas"
+COUCHDB_REPLICAS="$(kubectl -n "${COUCHDB_NAMESPACE}" get statefulset "${COUCHDB_STATEFULSET_NAME}" -o jsonpath='{.status.replicas}')"
+
+# If not supplied read set of latest snapshots
+if [ "${COUCHDB_SOURCE_SNAPSHOTS}" = '' ]; then
+  print_header "Reading latest set of snapshots"
+  read_latest_couchdb_snapshots
+fi
+
+# Check that number of snaphsots matches number of replicas
+[ "$(echo "${COUCHDB_SOURCE_SNAPSHOTS}" | wc -w | awk '{$1=$1};1')" = "${COUCHDB_REPLICAS}" ] \
+  || fail "Number of CouchDB snapshots (${COUCHDB_SOURCE_SNAPSHOTS}) does not match number of replicas (${COUCHDB_REPLICAS})"
+print "Snapshots to restore from: ${COUCHDB_SOURCE_SNAPSHOTS}"
+
+# Check snapshots
+print_header "Checking snapshots availability"
+check_snapshots
+
+# Scale services to 0
+print_header "Scaling down services"
+for d in $DEPLOYMENTS_NAMES; do
+  scale_and_wait deployment "${d}" "${DEPLOYMENTS_NAMESPACE}" 0
+done
+
+# Read names of current Couch volumes
+print_header "Reading CouchDB volume names"
+COUCHDB_PVS="$(kubectl -n "${COUCHDB_NAMESPACE}" get pvc -l "${COUCHDB_PVC_FILTER}" \
+                 --sort-by '.metadata.name' -o json | jq -re '.items[].spec.volumeName')"
+
+# Scale CouchDB to 0
+print_header "Scaling down CouchDB"
+scale_and_wait statefulset "${COUCHDB_STATEFULSET_NAME}" "${COUCHDB_NAMESPACE}" 0
+
+# Wait for pods to disappear
+print "Waiting for all pods to terminate"
+kubectl wait pods -l="${COUCHDB_POD_FILTER}" -n "${COUCHDB_NAMESPACE}" --for=delete --timeout="$((MAX_RETRIES * SLEEP))s"
+
+# Scale K8s snapshots to 0
+print_header "Scaling down k8s-snapshots"
+scale_and_wait deployment "${K8S_SNAPSHOTS_DEPLOYMENT_NAME}" "${K8S_SNAPSHOTS_NAMESPACE}" 0
+
+print_header "Restoring disks"
+# Restore individual CouchDB disks
+restore_disks
+
+# Scale CouchDB back
+print_header "Scaling up CouchDB"
+scale_and_wait statefulset "${COUCHDB_STATEFULSET_NAME}" "${COUCHDB_NAMESPACE}" "${COUCHDB_REPLICAS}"
+
+# Check CouchDB Health
+print_header "Checking CouchDB health"
+check_couchdb_health
+
+# Scale services back
```

Comment:
As the previous manual scenario, this script is missing DB consistency check. I mean – checking that CouchDB cluster has converged is good, but how do we know that actual DB data is in desired state? Maybe we should allow operator to check DB consistency manually (i.e. document counts and some predefined prefsets) and wait for the input before continuing to scale services back to restore flow of traffic?

Reply:
Yeah, that's a good idea, but let's handle it in a separate PR/ticket.

Reply:
@stepanstipl How about at least adding an instruction to check DB consistency before proceeding and wait for input, similarly to [this](https://github.com/gpii-ops/gpii-infra/blob/master/shared/rakefiles/scripts/kiali_ui.sh#L43-L47)?

Reply:
Done, unless we're in safe env (stg/dev) the script will now pause before scaling up the services and wait for user input.